dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/TimeoutTest.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class TimeoutTest {
+
+    public static final String APP_NAME = "timeoutTest";
+
+    @Server("timeoutServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        ShrinkHelper.defaultApp(server, APP_NAME, "timeoutTest");
+        FATSuite.copyAppsAppToDropins(server, APP_NAME);
+    }
+
+    @Test
+    public void testCheckpointStartTimeout() throws Exception {
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, server -> {
+            String result = server.waitForStringInLogUsingMark("TIMEOUT CALLED");
+            assertNotNull("Timeout call not found at checkpoint", result);
+        });
+        server.startServer();
+    }
+
+    @Test
+    public void testCheckpointTimeout() throws Exception {
```

Comment:
Order of tests will not be predictable, so it is possible `testCheckpointTimeout` will be called before `testCheckpointStartTimeout`.  I think you may need to place this in a single `@Test` method.

Also, I think to test this properly you need to have a delay between the checkpoint and the restore that is greater than the `@Timeout` annotations timeout value.